### DISCUSSION TITLE
Alerts and reports: Need docker-compose file

### DISCUSSION THREAD

**Discussion by JaiBie** [apache/superset#18013] [Q&A / Help] [answered]
Hi there,

just checked out superset and want to enable alerts and reports. To me it looks the documentation examples at superset.apache is older than  check out ones.

Can someone provide me with a working example based on docker-compose-non-dev version 3.7?

Thanks
Jai

**Comment by eschutho** [comment]
HI @JaiBie to use alerts and reports with docker, you'll need to overwrite the local config with the alerts and reports feature flag. Instructions for the local overrides are here: https://github.com/apache/superset/blob/master/docker/README.md#local-overrides
and then the flag that you'll want to enable is `ALERT_REPORTS`. This is the entire dictionary you'll want to add:
```
FEATURE_FLAGS = {
    "ALERT_REPORTS": True,
}
```
You can find more info here: https://github.com/apache/superset/blob/master/RESOURCES/FEATURE_FLAGS.md#in-testing

**Reply by hugopvl** [reply]
Hello, I was also trying to use Alerts & Reports, I changed the "ALERT_REPORTS" flag: True,
I did the smtp configuration, but the email is not sent.

but it also doesn't give any kind of error!

Is some more configuration needed?


![image](https://user-images.githubusercontent.com/81326296/149509038-2a18162a-ec50-4000-b702-8d7d0b55dd36.png)


![image](https://user-images.githubusercontent.com/81326296/149508577-6a5cb933-2c53-47f8-84d9-aabc26edaf40.png)


![image](https://user-images.githubusercontent.com/81326296/149508317-f5fb184d-f5f3-446c-8d8b-9ba14acf81be.png)

**Comment by JaiBie** [comment]
Hi @eschutho,

I already enabled feature flag. If i run docker-compose without daemon-mode I can see in console output superset generates report but mail is not being send out.
According to superset documentation there should also be configuration settings made at docker-compose file but example settings completely differs from actual docker-compose thats delivered with superset.

Or do I miss something else in configuration?

**Comment by JaiBie** [comment, marked as answer]
Ahhh, solved my problem - It was my stupid mistake :(

Set "ALERT_REPORTS_NOTIFICATION_DRY_RUN=True" this causes not sending mail.

After changing to False everything works. Maybe this is the same error that @hugopvl is facing.

**Reply by eschutho** [reply]
Great, glad you got it working!

**Reply by hugopvl** [reply]
> Great, glad you got it working!

Good afternoon, I really don't understand what else I can do, I've been dealing with this for days and there's no way it works.

I followed the steps several times and in fact it doesn't give any error but it doesn't send the email.

Any idea ? 

![image](https://user-images.githubusercontent.com/81326296/150194836-8704e7c4-3b16-4f4a-827e-d3885baeda3c.png)

**Reply by eschutho** [reply]
@hugopvl it looks like you're also in dry run mode. Have you tried what JaiBie did and add `ALERT_REPORTS_NOTIFICATION_DRY_RUN=False` to your configs?

**Reply by hugopvl** [reply]
> @hugopvl it looks like you're also in dry run mode. Have you tried what JaiBie did and add `ALERT_REPORTS_NOTIFICATION_DRY_RUN=False` to your configs?

yes, I already made that change, sorry for just replying now, but now it gives me another error: [Errno 99] Cannot assign requested address

![image](https://user-images.githubusercontent.com/81326296/151041149-723df072-5f76-47d0-b2bf-c60ded346fa0.png)
![image](https://user-images.githubusercontent.com/81326296/151041221-4c927f74-fe63-4c5d-be3b-ec8ced09a5d8.png)


it's not easy :/

**Reply by eschutho** [reply]
@hugopvl can you confirm in a browser that you can reach superset from the address shown in the line `Taking a PNG screenshot of url:`? From the error message, it looks like that address is unavailable.
